src/libcrun/utils.h

```diff
@@ -158,4 +158,23 @@ int get_file_type_fd (int fd, mode_t *mode);
 
 int str2sig (const char *name);
 
+#ifndef RESOLVE_NO_XDEV
+# define RESOLVE_NO_XDEV		0x01
+#endif
+#ifndef RESOLVE_NO_MAGICLINKS
+# define RESOLVE_NO_MAGICLINKS		0x02
+#endif
+#ifndef RESOLVE_NO_SYMLINKS
+# define RESOLVE_NO_SYMLINKS		0x04
+#endif
+#ifndef RESOLVE_BENEATH
+# define RESOLVE_BENEATH		0x08
+#endif
+#ifndef RESOLVE_IN_ROOT
+# define RESOLVE_IN_ROOT		0x10
+#endif
+
```

Comment:
I would add
```C
#ifndef __NR_openat2
# define __NR_openat2 437
#endif
```

as

1. chances are pretty high the code is going to be compiled with old(er) headers
2. the syscall number appears to be the same for all platforms except alpha which we prob don't care about
3. you already cared to provide the `RESOLVE_*` defines

Reply:
4. you can drop the `#ifdef` and the "else" part from `syscall_openat2` code.